### ISSUE TITLE
Grafana's `$__rate_interval` not compatible with GCP's Managed Prometheus

### ISSUE THREAD

**Issue by clupprich** [GoogleCloudPlatform/prometheus-engine#247]
We're running into an issue that's been reported to Grafana: https://github.com/grafana/grafana/issues/44542

The tl;dr of that is that in newer versions of Grafana, `$__rate_interval` can be formatted like `1h30m` (where it previously was `90m`). That issue/feature seems to be included in Prometheus 2.22.0 and up (https://github.com/grafana/grafana/issues/44251).

Unfortuantely, Managed Prometheus on GCP shows that issue and breaks dashboards in Grafana 8.3.4.

**Comment by lyanco** [comment]
Our query engine doesn't handle compound time units correctly right now. We'll have to add support for that.

In the meantime, lopping off the 30m or converting to 90m as you said will work.

Thanks for flagging this!

**Comment by clupprich** [comment]
@lyanco Just note that this can't be done in Grafana - there's no option to change the way how `$__rate_interval` gets formatted. So one would need to replace all uses of `$__rate_interval` and hardcode them, which means graphs will likely be broken when you zoom in/out.

For those interested, the change was introduced in https://github.com/grafana/grafana/pull/44035/commits/cbe6c1396f93c98073cf33d935b430311d89c47b: Previously, the rate interval was truncated/rounded by using [`intervalv2.FormatDuration`](https://github.com/grafana/grafana/blob/main/pkg/tsdb/intervalv2/intervalv2.go#L130-L157). Now it's going through [`Duration.String`](https://pkg.go.dev/time#example-Duration.String), which returns the exact representation of the duration.

**Comment by lyanco** [comment]
Yeah - figured. We end up playing catchup when Grafana decides to simply change how they do things on us.

Will look into this and see if we can't fix it sooner rather than later.

**Comment by clupprich** [comment]
Awesome, thanks @lyanco!

**Comment by lyanco** [comment]
FWIW I think downgrading your Grafana should work in the interim - 8.3.4 or 8.3.2 seem to work fine.

**Comment by clupprich** [comment]
Yeah, that would certainly work. Thanks for the tip!

**Comment by nokernel** [comment]
And sadly using 8.3.x version is not an option, we are using IAP proxy with grafana with auto_signup which got added in grafana 8.4.0 version.

Having this bug fixed would be very great, speficly since this breaks alot of pre-provisioned dashboards from prometheus-operator, when you use self deployed metrics scraping ( to not have to create new rules by hands ).

**Comment by birdx0810** [comment]
> FWIW I think downgrading your Grafana should work in the interim - 8.3.4 or 8.3.2 seem to work fine.

@lyanco 8.3.2 works, but not for 8.3.4 unfortunately. 

Another issue I realized is that queries with `[$interval:$resolution]` would fail in Grafana (tried `7.5.9`, `8.3.2`, `8.4.6`). 

Example Query (derived from the `kube-prometheus-stack` helm chart):
```
sum(irate(container_network_receive_bytes_total{cluster="$cluster",namespace=~"$namespace", pod=~"$pod"}[$interval:$resolution]))
```

Error raised:
```
bad_data: invalid parameter "query": substitute queries failed: irate() can only be used on a matrix selector (e.g. requests_total[5m])
```

**Comment by clupprich** [comment]
That's perfect new for a Friday! Thanks @lyanco and team!
